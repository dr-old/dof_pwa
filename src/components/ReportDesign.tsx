import { FileDownloadRounded } from "@mui/icons-material";
import { Box, Button, Typography, useTheme } from "@mui/material";

interface ArcDesignProps {
  backgroundColor?: string;
  title?: string;
}

export default function ReportDesign({
  backgroundColor,
  title,
}: ArcDesignProps) {
  const theme = useTheme();

  return (
    <Box
      sx={{
        backgroundColor,
        padding: theme.spacing(2),
        borderRadius: theme.shape.borderRadius,
        minHeight: 320,
      }}>
      <Box
        sx={{
          display: "flex",
          flexDirection: "row",
          alignItems: "center",
          justifyContent: "space-between",
        }}>
        <Button
          variant="outlined"
          size="small"
          sx={{
            fontSize: "0.5rem",
            borderRadius: 20,
            textTransform: "none",
            paddingX: 2,
          }}>
          <Typography variant="caption">{title}</Typography>
        </Button>
        <Typography
          variant="subtitle2"
          sx={{
            color:
              theme.palette.mode === "dark"
                ? theme.palette.warning.main
                : theme.palette.text.primary,
          }}>
          /2024
        </Typography>
      </Box>
      <Box
        sx={{
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
          height: "100%", // Full height of the parent container
          minHeight: 200, // Optional: to maintain some height
        }}>
        <Button
          variant="outlined"
          size="small"
          endIcon={<FileDownloadRounded fontSize="small" />}
          sx={{
            fontSize: "0.5rem",
            borderRadius: 4,
            textTransform: "none",
            paddingX: 2,
            paddingY: 2,
          }}>
          <Typography variant="caption">Download Report</Typography>
        </Button>
      </Box>
    </Box>
  );
}
