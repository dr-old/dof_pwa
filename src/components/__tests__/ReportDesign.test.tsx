// src/components/__tests__/ReportDesign.test.tsx
import { render, screen } from "@testing-library/react";
import { ThemeProvider, createTheme } from "@mui/material/styles";
import ReportDesign from "../ReportDesign";
import "@testing-library/jest-dom";

// Create a custom theme to use in tests
const theme = createTheme({
  palette: {
    mode: "light",
    text: { primary: "#000" },
    warning: { main: "#ff9800" },
  },
});

const renderComponent = (props = {}) => {
  return render(
    <ThemeProvider theme={theme}>
      <ReportDesign {...props} />
    </ThemeProvider>
  );
};

describe("ReportDesign Component", () => {
  test("renders correctly", () => {
    const { asFragment } = renderComponent({
      title: "Report Title",
      backgroundColor: "#f0f0f0",
    });

    // Snapshot Test
    expect(asFragment()).toMatchSnapshot();

    // Check if title is rendered inside the button
    expect(screen.getByText("Report Title")).toBeInTheDocument();

    // Check if the download button is present
    expect(screen.getByText("Download Report")).toBeInTheDocument();
  });
});
