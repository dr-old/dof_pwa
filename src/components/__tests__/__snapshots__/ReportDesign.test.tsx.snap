// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportDesign Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-d9jnep"
  >
    <div
      class="MuiBox-root css-1ialerq"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-rmeete-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
        >
          Report Title
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 css-6tcopt-MuiTypography-root"
      >
        /2024
      </h6>
    </div>
    <div
      class="MuiBox-root css-10dymat"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-1ehpx2r-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
        >
          Download Report
        </span>
        <span
          class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeSmall css-155nyw6-MuiButton-endIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="FileDownloadRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 9H15V4c0-.55-.45-1-1-1h-4c-.55 0-1 .45-1 1v5H7.41c-.89 0-1.34 1.08-.71 1.71l4.59 4.59c.39.39 1.02.39 1.41 0l4.59-4.59c.63-.63.19-1.71-.7-1.71M5 19c0 .55.45 1 1 1h12c.55 0 1-.45 1-1s-.45-1-1-1H6c-.55 0-1 .45-1 1"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
